### DISCUSSION TITLE
[Question] About Searching for Headers

### DISCUSSION THREAD

**Discussion by FelipeRearden** [tadashi-aikawa/obsidian-another-quick-switcher#93] [Q&A] [answered]
Hello @tadashi-aikawa !!!!

I am here thinking about something that I wanna ask about ... forgive me if I miss something :)

Example: I have a Custom Search where the 3 options are ENABLE: Tag - Header - Link

⦿ When i want to search for notes based on TAGS, I use `#` → #tag_1

**My Question: is there a specific term like `:@` that can be used to search for files that has a specific header**

Example: I want to search for notes with the tag `NBA`  AND that has the header `Michael Jordan`   → #NBA :@Michael_Jordan 

ℹ️ In my mind we would have a syntax like this in this order:

Option 01 -  1️⃣ Filename 2️⃣Tag 3️⃣Header → lorem ipsum #tag1 @header_name
Option 02 - 1️⃣ Filename 2️⃣Tag → lorem ipsum #tag1
Option 03 -  2️⃣Tag 3️⃣Header → #tag1 @header_name
Option 04 -  1️⃣ Filename 3️⃣Header → lorem ipsum @header_name

---

Thanks for reading this :)

Have a great day!!!!

**Comment by tadashi-aikawa** [comment, marked as answer]
Hi, @FelipeRearden :)

> My Question: is there a specific term like :@ that can be used to search for files that has a specific header

No, there is not. We can search files by header without a specific term. However, it is needed if you want to search only by headers. 

In my case, I search files by "file names", "headers", "tags", and "links", but it does not specify any of them precisely.

**Comment by FelipeRearden** [comment]
100% understood !!!

Thank you very much @tadashi-aikawa !!!!

**Comment by Fred-Vatin** [comment]
Obsidian| OS                           | Plugin
------- | ---------------------------- | ------
1.5.12  | Win 10 x64 pro (french) 22H2 | v11.4.0

I can not seem to find headers on files with custom search.

![image](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/assets/22337329/b8d3091f-d815-4797-842e-7185db85c7ee)

Here, searching by tag works but headers are not returned.
![Gif Animate 2024-05-08 04h52 #3616 _optimized](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/assets/22337329/f3b6eccb-ab2c-4774-8a3f-6d45750ad7b9)

**Edit** : after I installed ripgrep, headers are shown
![image](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/assets/22337329/0f914da7-f2b9-4a68-a113-82e1747a7e80)

I don’t think it is mentioned in the readme.md that ripgrep is also required for this feature to work.

**Reply by tadashi-aikawa** [reply]
@Fred-Vatin 

> I don’t think it is mentioned in the readme.md that ripgrep is also required for this feature to work.

That's so weird. There is no need to install ripgrep. 🤔

**Comment by Fred-Vatin** [comment]
I get it. It was not related to ripgrep. Actually, it doesn’t display headers from the note if this note is active (tab has focus)
